purchase_work_acceptance/models/stock_picking.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+from odoo.exceptions import ValidationError
+
+
+class Picking(models.Model):
+    _inherit = "stock.picking"
+
+    wa_id = fields.Many2one(
+        comodel_name='work.acceptance',
+        string='WA Reference',
+        domain=[('state', '=', 'accept')],
+    )
+
+    @api.multi
+    def button_validate(self):
+        self.ensure_one()
+        if self.wa_id:
+            wa_lines = self.env['work.acceptance.line']
+            for move in self.move_ids_without_package:
+                id = move.purchase_line_id.id
+                wa_line = wa_lines.search([('purchase_line_id', '=', id)])
```

Comment:
Can we have more than 1 wa_line with same `purchase_line_id` ???
This can be a BUG?

Reply:
Note:
* Beware of number of records can be on the result set, and make the variable name explicit. Normally, search will result in multi.
* If you want to use serach, and you expect the result to have len() = 1. Always use .ensure_one() after it. Or use search with limit = 1.